### PR TITLE
chore: cleanup unused autopilot vars

### LINKED ISSUE #1168: Remove unused autopilot variables
### TL;DR

Looks like we have a few unused vars in autopilot cluster modules we should cleanup like `kalm_config`

### Expected behavior

_No response_

### Observed behavior

_No response_

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
N/A
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by bharathkkb**
fixes #1168

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/beta-autopilot-private-cluster/main.tf [hunk 1]
```diff
@@ -151,12 +122,12 @@ locals {
   }]
   # BETA features
   cluster_istio_enabled                    = !local.cluster_output_istio_disabled
-  cluster_cloudrun_enabled                 = var.cloudrun
   cluster_dns_cache_enabled                = var.dns_cache
   cluster_telemetry_type_is_set            = var.cluster_telemetry_type != null
   cluster_pod_security_policy_enabled      = local.cluster_output_pod_security_policy_enabled
   cluster_intranode_visibility_enabled     = local.cluster_output_intranode_visbility_enabled
   cluster_vertical_pod_autoscaling_enabled = local.cluster_output_vertical_pod_autoscaling_enabled
+  cluster_gce_pd_csi_config                = var.gce_pd_csi_driver ? [{ enabled = true }] : [{ enabled = false }]
```
Comment (morgante) [inline modules/beta-autopilot-private-cluster/main.tf:130]: This is always enabled in autopilot: https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/gce-pd-csi-driver

We should either not make it a variable at all, or make it a variable which only accepts `true`.

File: autogen/main/outputs.tf.tmpl [hunk 2]
```diff
@@ -202,3 +197,10 @@ output "tpu_ipv4_cidr_block" {
   value       = var.enable_tpu ? google_container_cluster.primary.tpu_ipv4_cidr_block : null
 }
 {% endif %}
+{% if beta_cluster and  autopilot_cluster != true %}
```
[the remark this answers is not in the document]
Comment (bharathkkb) [reply autogen/main/outputs.tf.tmpl]: Done

File: autogen/main/outputs.tf.tmpl [hunk 3]
```diff
@@ -202,3 +197,10 @@ output "tpu_ipv4_cidr_block" {
...
+
+output "cloudrun_enabled" {
+  description = "Whether CloudRun enabled"
+  value       = local.cluster_cloudrun_enabled
```
[the remark this answers is not in the document]
Comment (bharathkkb) [reply autogen/main/outputs.tf.tmpl]: Hardcoded to false

File: modules/beta-autopilot-private-cluster/main.tf (hunk 1 above)
Comment (bharathkkb) [reply modules/beta-autopilot-private-cluster/main.tf]: Good catch! I did not make it a variable since users wont need to change this.

**Review by morgante — APPROVED** [review]

**Pull request merged**